pkg/controller/sub_controller/fe/prepare_modify.go

```diff
@@ -163,62 +137,14 @@ func (fc *Controller) dropObserverFromSqlClient(ctx context.Context, k8sclient c
 				podMap[item.Status.PodIP] = item.GetName()
 			}
 		}
-		frontendMap, err = buildSeqNumberToFrontend(allObserves, podMap, podTemplateName)
+		frontendMap, err = mysql.BuildSeqNumberToFrontendMap(allObserves, podMap, podTemplateName)
 		if err != nil {
 			klog.Errorf("DropObserverFromSqlClient failed, buildSeqNumberToFrontend err:%s", err.Error())
 			return nil
 		}
 	}
-	observes := getFirstFewFrontendsAfterDescendOrder(frontendMap, realScaleNumber)
+	observes := mysql.FindNeedDeletedFrontends(frontendMap, realScaleNumber)
```

Comment:
realScaleNumber the name is not clear for reading. It means needs be deleted number, so name is `needRemovedAmount` or directly use the shortly name "nra" and add comments to describe it.